desktop/cmp/persistenceManager/PersistenceManager.ts

```diff
@@ -0,0 +1,214 @@
+import {groupBy, keys, sortBy} from 'lodash';
+import {ReactNode} from 'react';
+import {manageDialog} from './impl/ManageDialog';
+import {saveDialog} from './impl/SaveDialog';
+import './PersistenceManager.scss';
+import {PersistenceManagerModel} from './PersistenceManagerModel';
+
+interface PersistenceManagerModelProps extends HoistProps {
+    model?: PersistenceManagerModel;
+    /** True to disable options for saving/managing items. */
+    minimal?: boolean;
+}
+
+export const [PersistenceManager, persistenceManager] =
+    hoistCmp.withFactory<PersistenceManagerModelProps>({
+        displayName: 'PersistenceManager',
+        model: uses(PersistenceManagerModel),
+
+        render({model, minimal = false}) {
+            const {
+                selectedObject,
+                isShared,
+                capitalPluralNoun,
+                manageDialogModel,
+                saveDialogModel
+            } = model;
+
+            return fragment(
+                hbox({
+                    className: 'persistence-manager',
+                    items: [
+                        popover({
+                            item: button({
+                                text: selectedObject?.name
+                                    ? getHierarchyDisplayName(selectedObject.name)
+                                    : capitalPluralNoun,
+                                icon: isShared ? Icon.users() : Icon.bookmark(),
+                                rightIcon: Icon.chevronDown(),
+                                outlined: true
+                            }),
+                            content: div({
+                                items: [
+                                    div({className: 'xh-popup__title', item: capitalPluralNoun}),
+                                    objMenu({minimal})
+                                ]
+                            }),
+                            placement: 'bottom-start'
+                        }),
+                        saveButton()
+                    ]
+                }),
+                manageDialogModel ? manageDialog({key: manageDialogModel.xhId}) : null,
+                saveDialogModel ? saveDialog({key: saveDialogModel.xhId}) : null
+            );
+        }
+    });
+
+//------------------------
+// Implementation
+//------------------------
+
+const saveButton = hoistCmp.factory<PersistenceManagerModel>({
+    render({model}) {
+        return button({
+            icon: Icon.save(),
+            tooltip: `Save changes to this ${model.noun}`,
+            intent: 'primary',
+            omit: !model.enableTopLevelSaveButton || !model.canSave,
+            onClick: () => model.saveAsync(null).linkTo(model.loadModel)
+        });
+    }
+});
+
+const objMenu = hoistCmp.factory<PersistenceManagerModelProps>({
+    render({model, minimal}) {
+        const {pluralNoun, objects, loadModel} = model,
+            grouped = groupBy(objects, it => it.group),
```

Comment:
Nit - could use the string property name as the iteratee arg to `groupBy` (i.e. `groupBy(objects, 'group')`)